File: Challenge #2/typescript/solution.ts

const mapper: object = {
  a: "4",
  b: "l3",
  c: "[",
  d: ")",
  e: "3",
  f: "|=",
  g: "&",
  h: "#",
  i: "1",
  j: ",_l",
  k: ">|",
  l: "1",
  m: "/\\/\\",
  n: "^/",
  o: "0",
  p: "|*",
  q: "(_,)",
  r: "l2",
  s: "5",
  t: "7",
  u: "|_|",
  v: "\/",
  w: "\/\/",
  x: "><",
  y: "`/",
  z: "2"
}

const transformNaturalLanguageToLeetLanguage = (text: string): void => {
  let leetText: string[] = []

  text.split('').map((word: string) => {
    const leetWord = mapper[word]
    if (leetWord === undefined) {
      leetText.push(' ')
    }
    leetText.push(leetWord)
  })

  console.log(leetText.join(''))
}

transformNaturalLanguageToLeetLanguage('This is a test language')

File: Challenge #2/python/solution.py
def transform_natural_language_to_leet_language(text):
    leet_mapper = {"a": "4", "b": "I3", "c": "[", "d": ")", "e": "3", "f": "|=", "g": "&", "h": "#", "i": "1",
                   "j": ",_|", "k": ">|", "l": "1", "m": "/\/\\", "n": " ^/", "o": "0", "p": " |*", "q": "(_,)",
                   "r": "I2", "s": "5", "t": "7", "u": "(_)", "v": "\/", "w": "\/\/", "x": "><", "y": "j", "z": "2"}
    leet_text = ""

    for word in text:
        if word.lower() in leet_mapper.keys():
            leet_text += leet_mapper[word.lower()]
        else:
            leet_text += word

    return leet_text


print(transform_natural_language_to_leet_language('this is a test language'))
